### PR TITLE
Speed up copy! by using memmove

### PR DESCRIPTION
**Pull request by hall-alex**
Take care of a TODO to replace a manual copy of elements (via a loop) with a call to memmove. 

For completeness: the TODO was for memcpy, but memmove is preferable, since it can handle overlaps. For reference, see also how [memmove is used in Julia's array.jl](https://github.com/JuliaLang/julia/blob/master/base/array.jl#L247):

### CODE REVIEW

**Comment by hall-alex** [comment]
Unsure, looks like for Julia 0.7 certain version requirements for MacroTools cannot be fulfilled ([making the build fail](https://travis-ci.org/github/RelationalAI-oss/Blobs.jl/jobs/709262136)).

```
┌ Warning: julia version requirement for project not satisfied
└ @ Pkg.Operations /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v0.7/Pkg/src/Operations.jl:339
ERROR: Unsatisfiable requirements detected for package MacroTools [1914dd2f]:
 MacroTools [1914dd2f] log:
 ├─possible versions are: [0.4.3-0.4.5, 0.5.0-0.5.5] or uninstalled
 ├─restricted to versions 0.5.0-0.5 by an explicit requirement, leaving only versions 0.5.0-0.5.5
 └─restricted by julia compatibility requirements to versions: 0.4.3-0.4.5 or uninstalled — no versions left
```

**Comment by rbvermaa** [comment]
I think we should remove 0.7 from the build. 1.0 is the LTS, so supporting that would make sense, but not 0.7 IMHO.

**Comment by rbvermaa** [comment]
Pushed a change to build against 1.0, 1.4 and nightly: ead7b998aad77a369cdbbad37c64a4bd41bcc0b6

**Comment by hall-alex** [comment]
Thanks!

File: src/vector.jl [hunk 1]
```diff
@@ -55,11 +54,12 @@ function Base.copy!(
...
         end
     end
-    if doff < soff
-        @inbounds for i in 0:n-1 dest[doff+i] = src[soff+i] end
-    else
-        @inbounds for i in n-1:-1:0 dest[doff+i] = src[soff+i] end
-    end
+    # Use memmove for speedy copying. Note: this correctly handles overlapping regions.
+    blob_size = Blobs.self_size(T)
+    ccall(:memmove, Ptr{Cvoid}, (Ptr{Cvoid}, Ptr{Cvoid}, Csize_t),
```
Comment (rgankema) [inline src/vector.jl:59]: Small style comment: according to RAIStyle (https://github.com/RelationalAI/RAIStyle#method-definitions), method calls and definitions that exceed the line limit, should be formatted like so

```
foo(
    long_argument_name_1,
    long_argument_name_2,
    long_argument_name_3,
    etc
)
````

Comment (hall-alex) [reply src/vector.jl]: Thanks, done.

**Pull request merged**